pymmmc/tests/clv/models/test_beta_geo.py

```diff
@@ -0,0 +1,205 @@
+            T=self.T,
+            a_prior=a_prior,
+            b_prior=b_prior,
+            alpha_prior=alpha_prior,
+            r_prior=r_prior,
+        )
+
+        assert isinstance(
+            model.model["a"].owner.op,
+            pm.HalfFlat if a_prior is None else type(a_prior.owner.op),
+        )
+        assert isinstance(
+            model.model["b"].owner.op,
+            pm.HalfFlat if b_prior is None else type(b_prior.owner.op),
+        )
+        assert isinstance(
+            model.model["alpha"].owner.op,
+            pm.HalfFlat if alpha_prior is None else type(alpha_prior.owner.op),
+        )
+        assert isinstance(
+            model.model["r"].owner.op,
+            pm.HalfFlat if r_prior is None else type(r_prior.owner.op),
+        )
+        assert model.model.eval_rv_shapes() == {
+            "a": (),
+            "a_log__": (),
+            "b": (),
+            "b_log__": (),
+            "alpha": (),
+            "alpha_log__": (),
+            "r": (),
+            "r_log__": (),
+        }
+
+    def test_model_convergence(self):
+        # b parameter has the largest mismatch of the four parameters
+        # rtol=0.3 should be enough
+        fit = self.fixed_model.fit_result.posterior
+        np.testing.assert_allclose(
+            [fit["a"].mean(), fit["b"].mean(), fit["alpha"].mean(), fit["r"].mean()],
+            [self.a_true, self.b_true, self.alpha_true, self.r_true],
+            rtol=0.3,
+        )
+
+    def test_expected_probability_alive(self):
+        """
+        The "true" prefix refers to the value obtained using 1) the closed form
+        solution and 2) the data-generating parameter values.
+        """
+        true_prob_alive = self.alive.mean()  # scalar
+        est_prob_alive = self.fixed_model.expected_probability_alive(
+            self.frequency,
+            self.recency,
+            self.T,
+        )
+
+        assert est_prob_alive.shape == (1, 1000, 500)
+        assert est_prob_alive.dims == ("chain", "draw", "customer_id")
+
+        np.testing.assert_allclose(
+            true_prob_alive.mean(),
+            est_prob_alive.mean(),
+            rtol=0.05,
+        )
+
+    def test_expected_num_purchases(self):
+        """
+        TODO: should we combine this test and the one below?
+        """
+
+        def to_xarray(array):
+            return xr.DataArray(
+                array,
+                coords={"customer_id": range(len(array))},
+                dims=("customer_id",),
+            )
+
+        t = to_xarray(range(20, 40, 2))
+        frequency = to_xarray([1, 3, 5, 7, 9] * 2)
+        recency = to_xarray([20, 30] * 5)
+        T = to_xarray([25, 35] * 5)
+        true_num_purchases = [
+            1.42554638,
+            2.17059763,
+            3.29959668,
+            4.06284244,
+            4.72380438,
+            1.73223875,
+            3.1645582,
+            4.08454744,
+            5.20603817,
+            6.34220761,
+        ]
```

Comment:
Where are you getting these numbers? Can you generate them in the `setup_class` method?

Reply:
Yes! I am now using the function provided in `BetaGeoFitter` from the `lifetimes` package